### PR TITLE
Week 3 joseph han

### PR DESCRIPTION
**Pull request by jhan-dev**
# Submitting Week 3 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [✅ ] Coding Challenges
- [✅ ] Interview Questions
- [✅ ] React Debugging

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -9,59 +12,238 @@
...
+  });
+});
+
+//After running yarn jest, I should expect a good fail prompt saying function is not defined.
+//    ReferenceError: sortObj is not defined
+
+//ALTERNATIVE SOLUTION FOR #2 USING Object.values()
+// const sortObj = (obj) => {
+//   //Return using Object.values() method using the passed in parameter obj, then using the sort() method a - b to sort all the elements inside from least to greatest
+//   return Object.values(obj).sort((a, b) => a - b)
```
Comment (elysemontano) [inline code-challenges.test.js:159]: I am a big fan of this particular approach as it is simpler and is not a for loop.

File: code-challenges.test.js [hunk 2]
```diff
@@ -9,59 +12,238 @@
...
+    expect(accuSum(accountTransactions3)).toEqual([]);
+
+  });
+});
+//After running yarn jest, I should expect a good fail prompt saying function is not defined.
+//    ReferenceError: accuSum is not defined
+
+//ALTERNATIVE SOLUTION FOR #3 USING .map() method
+// //We will create a single line arrow function return with no curly braces to use a .map() to iterate the arr parameter and at the start push 0 changes to the first iteration to the array copy output
+// const accuSum = (arr) => arr.map((currentVal = 0, addThis => currentval += addThis))
```
Comment (elysemontano) [inline code-challenges.test.js:223]: Really like that you are adding these alternate solutions!  Great work! 👏

File: interview-questions.md [hunk 3]
```diff
@@ -6,50 +6,91 @@ First, without external resources. Challenge yourself to answer from memory as i
...
+
 
 4. What is the difference between state values and props in React?
 
 Your answer:
+    //State values in REACT are local data that is stored within the component.
+    //Props are used to send data to a component to pass an argument.
 
 Researched answer:
+    //Both states and props are JavaScript objects. Both hold information that can determine the way something can render, they are different. Props get passsed to the component similar to a function getting a paramater as an input. While, state is controlled within the component it resides in.
```
Comment (elysemontano) [inline interview-questions.md:56]: State is not necessarily an object when using React hooks, but is just a special variable in React.

File: interview-questions.md [hunk 4]
```diff
@@ -6,50 +6,91 @@ First, without external resources. Challenge yourself to answer from memory as i
...
+
+
 
 5. What is the DOM?
 
 Your answer:
+    //Document-Object-Model
```
Comment (elysemontano) [inline interview-questions.md:63]: While correct, this answer in an interview is not very conversational and will likely not impress your interviewer.  Make sure to expand on your answers.

File: react-debugging.html [hunk 5]
```diff
@@ -65,11 +65,13 @@
         // No need to change anything ABOVE this line ^
 
         return (
+          //Wrapped fragment tags around return, added drawCard in line 71 onClick, fixed onclick to onClick on line 74.
```
Comment (elysemontano) [inline react-debugging.html:68]: Nice work! 💯

**Review by elysemontano — APPROVED** [review]
Great work Joseph!  I really like that you are trying multiple approaches to these problems!  If you want to continue to push yourself even more, consider possible edge cases that you can also test for.  A few of your answers in the interview section were still a little on the short side and won't give a good impression in an interview.  Keep working on expanding those answers.
